pkgs/applications/misc/bloodhound/default.nix

```diff
@@ -0,0 +1,127 @@
+{ lib
+, stdenv
+, fetchurl
+, makeWrapper
+, alsa-lib
+, at-spi2-atk
+, at-spi2-core
+, atk
+, cairo
+, cups
+, dbus
+, expat
+, fontconfig
+, freetype
+, gdk-pixbuf
+, glib
+, gnome2
+, gtk3
+, libGL
+, libappindicator-gtk3
+, libdrm
+, libnotify
+, libpulseaudio
+, libuuid
+, libxcb
+, libxkbcommon
+, libxshmfence
+, mesa
+, nspr
+, nss
+, pango
+, systemd
+, unzip
+, xdg-utils
+, xorg
+}:
+stdenv.mkDerivation rec {
+  pname = "bloodhound";
+  version = "4.3.1";
+
+  src = fetchurl {
+    url = "https://github.com/BloodHoundAD/BloodHound/releases/download/v${version}/BloodHound-linux-x64.zip";
+    sha256 = "sha256-OtQNrbPGAw5WoDeli6+REcx9ajor0CeF89L7Gg44aB0=";
+  };
+
+  rpath = lib.makeLibraryPath [
```

Comment:
No reason not to, I was just wondering if it was up-to-date as it was using the `rec` keyword and also mentioning quite old versions of NixOS such as `19.09`. 
The automatic patchelf hook fails, but I'll try the manual approach and once it's fixed, commit then squash.

Reply:
I'm getting an issue while using patchelf, I've removed the `--suffix LD_LIBRARY_PATH ${rpath}` and added to the `patchelf` line `--set-rpath ${finalAttrs.rpath}`, then I saw a missing library `libffmpeg.so`, so I added ` vivaldi-ffmpeg-codecs` that contains `libffmpeg.so` to the `rpath` variable.
```nix
  buildInputs = [
    gtk3 # needed for GSETTINGS_SCHEMAS_PATH
  ];

  nativeBuildInputs = [ makeWrapper ];

  dontBuild = true;
  dontPatchELF = false;

  installPhase = ''
    runHook preInstall
    ls -lah
    mkdir -p $out/{bin,lib,/lib/BloodHound}
    mv * $out/lib/BloodHound
    chmod +x $out/lib/BloodHound/BloodHound

    patchelf --set-interpreter ${stdenv.cc.bintools.dynamicLinker} \
       $out/lib/BloodHound/BloodHound --set-rpath ${finalAttrs.rpath}


    makeWrapper $out/lib/BloodHound/BloodHound $out/bin/BloodHound \
      --prefix XDG_DATA_DIRS : $GSETTINGS_SCHEMAS_PATH \
      --suffix PATH : ${lib.makeBinPath [xdg-utils]} \

    runHook postInstall
  '';
```
Running the binary now gives me this in dmesg : 
```
> [1]    3325649 trace trap (core dumped)  ./BloodHound
# in dmesg
[138298.025413] traps: ThreadPoolForeg[3328187] trap int3 ip:560215d73543 sp:7f5065e91060 error:0 in BloodHound[5602154dc000+6068000]
```
Trying to debug using gdb isn't helpful much as there is no debugging symbols, should I try to compile the binary with debugging symbols and investigate?

Reply:
> The automatic patchelf hook fails, but I'll try the manual approach and once it's fixed, commit then squash.

How does it fail?

Reply:
It fails, by saying it could not resolve 7 dependencies : 
```nix
auto-patchelf: 7 dependencies could not be satisfied
error: auto-patchelf could not satisfy dependency libnss3.so wanted by /nix/store/pp0agm1frrar1w7lk7ipp7qjkz2flf1s-bloodhound-4.3.1/lib/BloodHound/BloodHound
error: auto-patchelf could not satisfy dependency libnssutil3.so wanted by /nix/store/pp0agm1frrar1w7lk7ipp7qjkz2flf1s-bloodhound-4.3.1/lib/BloodHound/BloodHound
error: auto-patchelf could not satisfy dependency libsmime3.so wanted by /nix/store/pp0agm1frrar1w7lk7ipp7qjkz2flf1s-bloodhound-4.3.1/lib/BloodHound/BloodHound
error: auto-patchelf could not satisfy dependency libnspr4.so wanted by /nix/store/pp0agm1frrar1w7lk7ipp7qjkz2flf1s-bloodhound-4.3.1/lib/BloodHound/BloodHound
error: auto-patchelf could not satisfy dependency libdrm.so.2 wanted by /nix/store/pp0agm1frrar1w7lk7ipp7qjkz2flf1s-bloodhound-4.3.1/lib/BloodHound/BloodHound
error: auto-patchelf could not satisfy dependency libgbm.so.1 wanted by /nix/store/pp0agm1frrar1w7lk7ipp7qjkz2flf1s-bloodhound-4.3.1/lib/BloodHound/BloodHound
error: auto-patchelf could not satisfy dependency libasound.so.2 wanted by /nix/store/pp0agm1frrar1w7lk7ipp7qjkz2flf1s-bloodhound-4.3.1/lib/BloodHound/BloodHound
auto-patchelf failed to find all the required dependencies.
Add the missing dependencies to --libs or use `--ignore-missing="foo.so.1 bar.so etc.so"`.
/nix/store/blpvf60m29q02c0lc5fyhim30ma4y1vv-stdenv-linux/setup: line 87: pop_var_context: head of shell_variables not a function context
/nix/store/blpvf60m29q02c0lc5fyhim30ma4y1vv-stdenv-linux/setup: line 1459: pop_var_context: head of shell_variables not a function context
/nix/store/blpvf60m29q02c0lc5fyhim30ma4y1vv-stdenv-linux/setup: line 1596: pop_var_context: head of shell_variables not a function context
error: builder for '/nix/store/bci47w7wd4scax8hq5nhnaqjp87nbi1p-bloodhound-4.3.1.drv' failed with exit code 1;
       last 10 log lines:
       > error: auto-patchelf could not satisfy dependency libsmime3.so wanted by /nix/store/pp0agm1frrar1w7lk7ipp7qjkz2flf1s-bloodhound-4.3.1/lib/BloodHound/BloodHound
       > error: auto-patchelf could not satisfy dependency libnspr4.so wanted by /nix/store/pp0agm1frrar1w7lk7ipp7qjkz2flf1s-bloodhound-4.3.1/lib/BloodHound/BloodHound
       > error: auto-patchelf could not satisfy dependency libdrm.so.2 wanted by /nix/store/pp0agm1frrar1w7lk7ipp7qjkz2flf1s-bloodhound-4.3.1/lib/BloodHound/BloodHound
       > error: auto-patchelf could not satisfy dependency libgbm.so.1 wanted by /nix/store/pp0agm1frrar1w7lk7ipp7qjkz2flf1s-bloodhound-4.3.1/lib/BloodHound/BloodHound
       > error: auto-patchelf could not satisfy dependency libasound.so.2 wanted by /nix/store/pp0agm1frrar1w7lk7ipp7qjkz2flf1s-bloodhound-4.3.1/lib/BloodHound/BloodHound
       > auto-patchelf failed to find all the required dependencies.
       > Add the missing dependencies to --libs or use `--ignore-missing="foo.so.1 bar.so etc.so"`.
       > /nix/store/blpvf60m29q02c0lc5fyhim30ma4y1vv-stdenv-linux/setup: line 87: pop_var_context: head of shell_variables not a function context
       > /nix/store/blpvf60m29q02c0lc5fyhim30ma4y1vv-stdenv-linux/setup: line 1459: pop_var_context: head of shell_variables not a function context
       > /nix/store/blpvf60m29q02c0lc5fyhim30ma4y1vv-stdenv-linux/setup: line 1596: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/bci47w7wd4scax8hq5nhnaqjp87nbi1p-bloodhound-4.3.1.drv'
```
I did not understand, how to actually use the `--libs` inside the derivation in order to fix this issue.
[Full logs gist](https://gist.github.com/AkechiShiro/6b03c880e9b807a77078c58b9a385b09)

Reply:
Anything I can do to move forward the issue ? Any tips on fixing this issue ? I'd like to move the PR forward but I'm not sure what is the proper approach at the moment.

Reply:
I fixed it, it was missing `libudev.so` it seems, and it was not needed by the binary but adding it using `patchelf` fixed the issue.